controller/api/destination/server.go

```diff
@@ -206,57 +206,13 @@ func (s *server) GetProfile(dest *pb.GetDestination, stream pb.Destination_GetPr
 			if err != nil {
 				return err
 			}
-			var endpoint *pb.WeightedAddr
-			opaquePorts := make(map[uint32]struct{})
-			if pod != nil {
-				// If the IP maps to a pod, we create a single endpoint and
-				// return it in the DestinationProfile response
-				podSet := podToAddressSet(s.k8sAPI, pod).WithPort(port)
-				podID := watcher.PodID{
-					Namespace: pod.Namespace,
-					Name:      pod.Name,
-				}
-				var ok bool
-				opaquePorts, ok, err = getPodOpaquePortsAnnotations(pod)
-				if err != nil {
-					log.Errorf("failed to get opaque ports annotation for pod: %s", err)
-				}
-				// If the opaque ports annotation was not set, then set the
-				// endpoint's opaque ports to the default value.
-				if !ok {
-					opaquePorts = s.defaultOpaquePorts
-				}
-
-				skippedInboundPorts, err := getPodSkippedInboundPortsAnnotations(pod)
-				if err != nil {
-					log.Errorf("failed to get ignored inbound ports annotation for pod: %s", err)
-				}
-
-				endpoint, err = toWeightedAddr(podSet.Addresses[podID], opaquePorts, skippedInboundPorts, s.enableH2Upgrade, s.identityTrustDomain, s.controllerNS, log)
-				if err != nil {
-					return err
-				}
-				// `Get` doesn't include the namespace in the per-endpoint
-				// metadata, so it needs to be special-cased.
-				endpoint.MetricLabels["namespace"] = pod.Namespace
-			}
 
-			// When the IP does not map to a service, the default profile is
-			// sent without subscribing for future updates. If the IP mapped
-			// to a pod, then the endpoint will be set in the response.
-			translator := newProfileTranslator(stream, log, "", port, endpoint)
-
-			// If there are opaque ports then update the profile translator
-			// with a service profile that has those values
-			//
-			// TODO: Remove endpoint from profileTranslator and set it here
-			// similar to opaque ports
-			if len(opaquePorts) != 0 {
-				sp := sp.ServiceProfile{}
-				sp.Spec.OpaquePorts = opaquePorts
-				translator.Update(&sp)
-			} else {
-				translator.Update(nil)
+			// If the IP maps to a pod, we create a single endpoint and
```

Comment:
Note that the IP may or may not map to a pod (getPod might have returned nil) but in either case we want to return a single endpoint.

Reply:
I modified the comment here and expanded a bit (to first of mark the possibility of absence for the pod). I think if we do not have a pod we send a default profile which will not have an endpoint in, I tried to make it more clear in the comment. Let me know what you think!

Reply:
This surprised me, but I took a look and you're right.  I would have expected us to send the default profile but with the queried ip:port set as the endpoint.  I suppose this probably doesn't come up since we only do profile lookups for IPs in the cluster network which will almost certainly map to either a pod or service IP.